hunting/okta/docs/queries/initial_access_password_spraying_from_repeat_source.toml

```diff
@@ -0,0 +1,39 @@
+[hunt]
+author = "Elastic"
+description = """
+This hunting query identifies password spraying attacks in Okta where the same source IP attempts to authenticate to multiple accounts with invalid credentials. Adversaries may attempt to use a single source IP to avoid detection and bypass account lockout policies.
+"""
+integration = ["okta"]
+uuid = "1c2d2b08-71ee-11ef-952e-f661ea17fbcc"
+name = "Password Spraying from Repeat Source"
+language = ["ES|QL"]
+license = "Elastic License v2"
+notes = [
+    "`okta.actor.alternate_id` are the targeted accounts.",
+    "Pivot for successful logins from the same source IP by searching for `event.action` equal to `user.session.start` or `user.authentication.verify` where the outcome is `SUCCESS`.",
+    "User agents can be used to identify anomalous behavior, such as a user agent that is not associated with a known application or user.",
+    "The `target_count` can be adjusted depending on the organization's account lockout policy or baselined behavior."
+]
+mitre = ['T1078.004']
+query = ['''
+from logs-okta*
+| where @timestamp > NOW() - 7 day
+
+// truncate the timestamp to daily intervals
+| eval target_time_window = DATE_TRUNC(1 days, @timestamp)
+| where
+
+// filter for invalid credential events
+    event.action == "user.session.start"
+    and okta.outcome.result == "FAILURE"
+    and okta.outcome.reason == "INVALID_CREDENTIALS"
+| stats
+    // count the distinct number of targeted accounts
+    target_count = count_distinct(okta.actor.alternate_id),
+
+    // count the number of invalid credential events for each source IP
+    source_count = count(*) by target_time_window, okta.client.ip
+
+// filter for source IPs with more than 5 invalid credential events
+| where target_count >= 5
```

Comment:
in pwd spraying attacks they need to do so with 100 of accounts I suspect 5 is too low

Reply:
@Samirbous - The thing to consider is that there will only be a system log generated if the account they are attempting to login with is a valid account for that org. If not, then there is no log so if the same address has 5 login attempts where each user name is different in a 15-minute window, that is a bit strange for an IdP to report. Maybe to strengthen a bit, we can focus on user accounts specifically to ignore machine-to-machine or token related auth.

Reply:
pwd spraying suppose the adversary has already a list of valid users, so if it fail it should be logged, targeting only 5 in 15m is also suspicious, anyway its a hunt the users can adjust threshold to their own env 👍